core/src/main/java/com/oath/oak/OakRValueBufferImpl.java

```diff
@@ -182,14 +182,16 @@ public double getDouble(int index) {
         if (transformer == null) {
             throw new NullPointerException();
         }
-        Result<T> result = valueOperator.transform(getValueSlice(), transformer, version);
-        if (result.operationResult == FALSE) {
-            throw new ConcurrentModificationException();
-        } else if (result.operationResult == RETRY) {
-            lookupValueReference();
-            transform(transformer);
+        while (true) {
+            Result<T> result = valueOperator.transform(getValueSlice(), transformer, version);
```

Comment:
1) Why the PR name starts with "correctness"? :)
2) Please add an assertion for infinite loop